src/strategies/hedgey-multi/index.ts

```diff
@@ -0,0 +1,187 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { Multicaller } from '../../utils';
+
+export const author = 'Hedgey';
+export const version = '0.1.0';
+
+enum ContractType {
+  NFT = 'NFT',
+  TokenInfusedNFT = 'TokenInfusedNFT'
+}
+
+type ContractDetails = {
+  address: string;
+  token: string;
+  decimal: number;
+  contractType: ContractType;
+  multiplyer?: string;
+};
+
+const abis = {
+  NFT: [
+    'function balanceOf(address owner) external view returns (uint256 balance)',
+    'function tokenOfOwnerByIndex(address owner, uint256 index) external view returns (uint256 tokenId)',
+    'function futures(uint256 index) external view returns (uint256 amount, address token, uint256 unlockDate)',
+    'function token() external view returns (address token)'
+  ],
+  TokenInfusedNFT: [
+    'function futures(uint256 index) external view returns (uint256 amount, uint256 unlockDate)'
+  ]
+};
+
+const compareAddresses = (address1: string, address2: string): boolean => {
+  if (!address1 || !address2) return false;
+  return address1.toLowerCase() === address2.toLowerCase();
+};
+
+export async function strategy(
+  _space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const contractDetails: ContractDetails[] = options.contracts;
```

Comment:
Can we add a limit to `contracts` options? someone passes in unlimited contracts. better to throw error if someone pass more contract